### ISSUE TITLE
unable to mb run

### ISSUE THREAD

**Issue by JunhongYu** [CobraLab/MAGeTbrain#46]
Hi there,

i've installed the anaconda , pyminc and minc-stuff, but for some reason i'm unable to proceed with 'mb run' I have pasted the log below. Did i miss anything?

[gerard@sklfmri3 ROI]$ mb check
[gerard@sklfmri3 ROI]$ mb status
5 atlases, 21 templates, 288 subjects found
105 atlas-to-template registration commands left
6048 template-to-subject registration commands left
30240 transform merging commands left
30240 label propagation commands left
288 label fusion commands left
[gerard@sklfmri3 ROI]$ mb run
5 atlases, 21 templates, 288 subjects found
running 105 commands after stage NONE*
Traceback (most recent call last):
File "/home/gerard/MAGeTbrain/bin/qbatch", line 147, in 
o, e = execute("qsub %s" % scriptfile, dry_run = options.dry_run)
File "/home/gerard/MAGeTbrain/bin/qbatch", line 24, in execute
stdin = subprocess.PIPE, stdout = subprocess.PIPE , stderr = subprocess.PIPE)
File "/home/gerard/anaconda2/lib/python2.7/subprocess.py", line 390, in init
errread, errwrite)
File "/home/gerard/anaconda2/lib/python2.7/subprocess.py", line 1025, in _execute_child
raise child_exception
OSError: [Errno 2] No such file or directory
Traceback (most recent call last):
File "/home/gerard/MAGeTbrain/bin/mb", line 684, in 
main()
File "/home/gerard/MAGeTbrain/bin/mb", line 191, in main
options.func(options)
File "/home/gerard/MAGeTbrain/bin/mb", line 291, in command_run
p.run()
File "/home/gerard/MAGeTbrain/bin/mb", line 526, in run
self.queue.run(stages=self._get_stage_order())
File "/home/gerard/MAGeTbrain/bin/mb", line 477, in run
walltime=walltime, processors = processors)
File "/home/gerard/MAGeTbrain/bin/mb", line 487, in qbatch
input='\n'.join(commands))
File "/home/gerard/MAGeTbrain/bin/mb", line 438, in execute
raise Exception("Returns %i :: %s" %( proc.returncode, command ))
Exception: Returns 1 :: qbatch -N mb_templatelib_2017-12-01T14-40-49 --afterok_pattern NONE* - 4 6:00:00

**Comment by gdevenyi** [comment]
Do you have a PBS cluster? MAGeTbrain is by default configured to use PBS.

**Comment by JunhongYu** [comment]
i don't think so.  Is it possible to configure it to not use PBS?

**Comment by gdevenyi** [comment]
Hi,

As per the "--help":
```
Execution options:
  -q {parallel,qbatch,files}, --queue {parallel,qbatch,files}
                        Queueing method to use
```

You can use the "parallel" method to run with GNU Parallel.

Please note that MAGeTbrain is extremely computationally expensive and a cluster is highly recommended.

**Comment by JunhongYu** [comment]
it still doesnt work 

[gerard@sklfmri3 ROI]$ mb run -q parallel
5 atlases, 21 templates, 288 subjects found
WARNING: 'vanilla' mode only works for the 'qbatch' queue.
WARNING: Other queues treat this as stage as 'register'. You will need to run the 'voting' stage separately.
Traceback (most recent call last):
  File "/home/gerard/MAGeTbrain/bin/mb", line 684, in <module>
    main()
  File "/home/gerard/MAGeTbrain/bin/mb", line 191, in main
    options.func(options)
  File "/home/gerard/MAGeTbrain/bin/mb", line 291, in command_run
    p.run()
  File "/home/gerard/MAGeTbrain/bin/mb", line 526, in run
    self.queue.run(stages=self._get_stage_order())
  File "/home/gerard/MAGeTbrain/bin/mb", line 452, in run
    self.parallel(self.commands[stage])
  File "/home/gerard/MAGeTbrain/bin/mb", line 458, in parallel
    self.execute(command, input='\n'.join(commands))
  File "/home/gerard/MAGeTbrain/bin/mb", line 435, in execute
    stdin = subprocess.PIPE, stdout = 2, stderr = 2)
  File "/bcs/projects/gerard_ADNI/ROI/anaconda2/lib/python2.7/subprocess.py", line 390, in __init__
    errread, errwrite)
  File "/bcs/projects/gerard_ADNI/ROI/anaconda2/lib/python2.7/subprocess.py", line 1025, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory

i've tried -q files too, and it doesn't do anything at all.

**Comment by gdevenyi** [comment]
```
WARNING: 'vanilla' mode only works for the 'qbatch' queue.
```

```sh
> mb run register -q parallel
> mb run vote -q parallel
```

Please take a close look over the the ``mb run --help`` options, otherwise you may overload your machine with too many processes.

**Comment by JunhongYu** [comment]
it still gives the same error. I figure it may have something to do with the subprocess.py file. But i've check that file is located exactly where it should be, and i have read and write permissions on that folder.

i dont know if this is related, but when i do run the subprocess.py on its own, i got this 

[gerard@sklfmri3 python2.7]$ python subprocess.py
Process list:
   PID TTY          TIME CMD
 28857 pts/7    00:00:00 bash
 29126 pts/7    00:00:00 python
 29127 pts/7    00:00:00 ps

Looking for 'hda'...
''

Trying a weird file...
The file didn't exist.  I thought so...
Child traceback:
Traceback (most recent call last):
  File "subprocess.py", line 976, in _execute_child
    os.execvp(executable, args)
  File "/bcs/projects/gerard_ADNI/anaconda2/lib/python2.7/os.py", line 346, in execvp
    _execvpe(file, args)
  File "/bcs/projects/gerard_ADNI/anaconda2/lib/python2.7/os.py", line 370, in _execvpe
    func(file, *argrest)
OSError: [Errno 2] No such file or directory

**Comment by gdevenyi** [comment]
Can I see the full command you're running and the error stream for ``mb run register -q parallel``

**Comment by JunhongYu** [comment]
[gerard@sklfmri3 ROI]$ mb check
[gerard@sklfmri3 ROI]$ mb status
5 atlases, 21 templates, 288 subjects found
105 atlas-to-template registration commands left
6048 template-to-subject registration commands left
30240 transform merging commands left
30240 label propagation commands left
288 label fusion commands left
[gerard@sklfmri3 ROI]$ mb run register -q parallel
5 atlases, 21 templates, 288 subjects found
Traceback (most recent call last):
  File "/home/gerard/MAGeTbrain/bin/mb", line 684, in <module>
    main()
  File "/home/gerard/MAGeTbrain/bin/mb", line 191, in main
    options.func(options)
  File "/home/gerard/MAGeTbrain/bin/mb", line 291, in command_run
    p.run()
  File "/home/gerard/MAGeTbrain/bin/mb", line 526, in run
    self.queue.run(stages=self._get_stage_order())
  File "/home/gerard/MAGeTbrain/bin/mb", line 452, in run
    self.parallel(self.commands[stage])
  File "/home/gerard/MAGeTbrain/bin/mb", line 458, in parallel
    self.execute(command, input='\n'.join(commands))
  File "/home/gerard/MAGeTbrain/bin/mb", line 435, in execute
    stdin = subprocess.PIPE, stdout = 2, stderr = 2)
  File "/bcs/projects/gerard_ADNI/anaconda2/lib/python2.7/subprocess.py", line 390, in __init__
    errread, errwrite)
  File "/bcs/projects/gerard_ADNI/anaconda2/lib/python2.7/subprocess.py", line 1025, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory

**Comment by gdevenyi** [comment]
Do you have minc-toolkit installed?
Do you have gnu-parallel installed?

This is a failure indicating that the program magetbrain is trying to run doesn't.

**Comment by JunhongYu** [comment]
i have previously installed minc-toolkit 1.9.15 ,  and i have just installed gnu parallel, so now i got a different error:

/home/gerard/MAGeTbrain/bin/mb_register: line 34: mincANTS: command not found

should i install minc-toolkit 1.0.07 instead?

**Comment by gdevenyi** [comment]
Newer versions renamed mincANTS to ANTS, please just make a copy of that binary in /opt/minc/1.9 15/bin

**Comment by gdevenyi** [comment]
You will also need minc-stuffs and pyminc from the mouse imaging center GitHub, please install those.

**Comment by JunhongYu** [comment]
i have installed pyminc and minc-stuff previously. So i've renamed ANTS to mincANTS and ran it again. Then i got several lines of "mincANTS: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.15' not found" 

 where can i get this GLIBCXX? i cant find it anywhere on my server

**Comment by gdevenyi** [comment]
That kind of an error indicates a mismatch in the global C library versions.

Do you have perahps multiple versions of minc-toolkit installed? Are you installing on a supported linux OS?

What OS and how/what do did you install for minc-toolkit?

**Comment by CyLouisKoo** [comment]
> it still gives the same error. I figure it may have something to do with the subprocess.py file. But i've check that file is located exactly where it should be, and i have read and write permissions on that folder.
> 
> i dont know if this is related, but when i do run the subprocess.py on its own, i got this
> 
> [gerard@sklfmri3 python2.7]$ python subprocess.py
> Process list:
> PID TTY TIME CMD
> 28857 pts/7 00:00:00 bash
> 29126 pts/7 00:00:00 python
> 29127 pts/7 00:00:00 ps
> 
> Looking for 'hda'...
> ''
> 
> Trying a weird file...
> The file didn't exist. I thought so...
> Child traceback:
> Traceback (most recent call last):
> File "subprocess.py", line 976, in _execute_child
> os.execvp(executable, args)
> File "/bcs/projects/gerard_ADNI/anaconda2/lib/python2.7/os.py", line 346, in execvp
> _execvpe(file, args)
> File "/bcs/projects/gerard_ADNI/anaconda2/lib/python2.7/os.py", line 370, in _execvpe
> func(file, *argrest)
> OSError: [Errno 2] No such file or directory

have you solved this problem?

**Comment by gdevenyi** [comment]
Hi,

Your error does not seem to be in any way related. Please provide a full description of how you installed, configured and ran your copy of MAGeTbrain *in a new issue* in order to get support.
